modules/auxiliary/gather/zoomeye_search.rb

```diff
@@ -1,178 +1,210 @@
 ##
 # This module requires Metasploit: https://metasploit.com/download
 # Current source: https://github.com/rapid7/metasploit-framework
+# rewrite zoomeye search module 2021/07/14
+#
 ##
 
 
-
 class MetasploitModule < Msf::Auxiliary
   include Msf::Auxiliary::Report
 
-
-  def initialize(info={})
-    super(update_info(info,
-      'Name'        => 'ZoomEye Search',
-      'Description' => %q{
-        The module use the ZoomEye API to search ZoomEye. ZoomEye is a search
-        engine for cyberspace that lets the user find specific network
-        components(ip, services, etc.).
-      },
-      'Author'      => [ 'Nixawk' ],
-      'References'  => [
-        ['URL', 'https://github.com/zoomeye/SDK'],
-        ['URL', 'https://www.zoomeye.org/api/doc'],
-        ['URL', 'https://www.zoomeye.org/help/manual']
-      ],
-      'License'     => MSF_LICENSE
-      ))
-
-      register_options(
-        [
-          OptString.new('USERNAME', [true, 'The ZoomEye username']),
-          OptString.new('PASSWORD', [true, 'The ZoomEye password']),
-          OptString.new('ZOOMEYE_DORK', [true, 'The ZoomEye dork']),
-          OptEnum.new('RESOURCE', [true, 'ZoomEye Resource Type', 'host', ['host', 'web']]),
-          OptInt.new('MAXPAGE', [true, 'Max amount of pages to collect', 1])
-        ])
+  def initialize(info = {})
+    super(update_info(
+            info,
+            'Name' => "ZoomEye Search",
+            "Description" => %q{
+            The module use the ZoomEye API to search ZoomEye.ZoomEye is a cyberspace search engine,
+            users can search for network devices using a browser.
+            },
+            'Author' => [
+              'Nixawk', # Original
+              'wh0am1i' # Rewrite
+            ],
+            "References" => [
+              ["URL", "https://www.zoomeye.org/api/doc"],
+              ["URL", "https://github.com/knownsec/ZoomEye-python"]
+            ],
+            "License" => MSF_LICENSE
+          ))
+    register_options(
+      [
+        OptString.new('APIKEY', [true, 'The ZoomEye API KEY']),
+        OptString.new('DORK', [true, 'The ZoomEye dork']),
+        OptEnum.new('RESOURCE', [true, 'ZoomEye Resource Type', 'host', ['host', 'web']]),
+        OptString.new('FACETS', [false, 'Query the distribution of the full data of the dork']),
+        OptInt.new('PAGE', [true, 'Max amount of pages to collect', 1]),
```

Comment:
Using `MAXPAGE` for consistency with the shodan module would be helpful.
```suggestion
        OptInt.new('MAXPAGE', [true, "Max amount of pages to collect", 1]),
```